src/CodeGen_OpenCL_Dev.cpp

```diff
@@ -16,6 +17,131 @@ using std::sort;
 
 static ostringstream nil;
 
+// OpenCL doesn't support vectors of bools, this mutator rewrites IR
+// to use signed integer vectors instead.
+class EliminateBoolVectors : public IRMutator {
+private:
+    using IRMutator::visit;
+
+    template <typename T>
+    void visit_comparison(const T* op) {
+        Expr a = mutate(op->a);
+        Expr b = mutate(op->b);
+
+        Type t = a.type();
+        if (t.width > 1) {
+            // To represent bool vectors, OpenCL uses vectors of signed
+            // integers with the same width as the types being compared.
+            t.code = Type::Int;
+            expr = T::make(a, b);
```

Comment:
The Param actually will get promoted to an int (it will get changed from a broadcast of a bool to a broadcast of an integer cast). However, you're right that the comparison ops should also widen both operands to the wider of the two types.